glean-core/src/lib.rs

```diff
@@ -163,7 +163,7 @@ pub struct Configuration {
 pub struct Glean {
     upload_enabled: bool,
     data_store: Option<Database>,
-    event_data_store: EventDatabase,
+    event_data_store: Option<EventDatabase>,
     core_metrics: CoreMetrics,
     internal_pings: InternalPings,
```

Comment:
@badboy @mdboom how do you both like this approach of creating a stub Glean object?

Creating a FFI API to only create a "PingUploader" object would require us to keep the "handle" alive like we do for the Glean object...

Note: I will need to make `core_metrics` and `internal_pings` optional as well.

Reply:
Not a fan. :P 

More constructive: We don't need any of these fields for the upload manager.
I think we should be able to only create the uploader object. It will be created on each process launch anyway.
That will change the FFI slightly:
* We need to store it (not the worst: put it in a global)
* We either:
	* need a special API to run against that global object
	* or an additional check in the current API to either check the global Glean object or the upload manager (then again, and this is just thinking, what if we remove the upload manager from the Glean object and _only_ have the global singleton of it? Would that make things easier or more complicated?) 

Reason I dislike making more things optional:
* Having a Glean object allocated (and thus "initialized" from the Python view) but only half-filled with stuff will make it hard to detect if we call any metric recording by accident (whereas if it is not initialized it will be a no-op or crash [which would be a bug])
* `Option`s are not free. It's another few bytes on top of our struct and requires an additional branch on each access (probably not super important, as there's only one of these objects, but still overhead for consumers thatn don't require the only-upload-mode)

Reply:
As much as it adds additional work, I think I agree with @badboy here.  It's *almost* as if the upload stuff could/should be completely isolated in its own crate (not suggesting we do that, but we should at least think of it that way, maybe).

Reply:
All right, good feedback. I reworked this a bit!